kong/router/atc.lua

```diff
@@ -469,7 +481,7 @@ function _M:exec(ctx)
   set_upstream_uri(req_uri, match_t)
 
   -- debug HTTP request header logic
-  add_debug_headers(var, header, match_t)
+  add_debug_headers(ctx, header, match_t)
```

Comment:
Could we change the logic of `add_debug_headers()` like this:
```
if get_header("kong_debug", ctx) then
  add_debug_headers(header, match_t)
end
```

Reply:
@chronolaw. the `add_debug_headers` function has this code at very top so I am not sure if your suggestion is good:
```lua
  if not kong.configuration.allow_debug_header then
    return
  end
  
  if not get_header("kong_debug", ctx) then
    return
  end
  ```

Reply:
I think that if we do so we can remove the usage of `ctx` ( or original `var`) in utils.lua.